src/decode/rangecoder.rs

```diff
@@ -28,9 +28,51 @@ where
         Ok(dec)
     }
 
+    pub fn from_parts(stream: &'a mut R, range: u32, code: u32) -> Self {
+        Self {
+            stream,
+            range,
+            code,
+        }
+    }
+
+    pub fn into_parts(self) -> (u32, u32) {
+        (self.range, self.code)
+    }
+
+    pub fn set(&mut self, range: u32, code: u32) {
+        self.range = range;
+        self.code = code;
+    }
+
+    pub fn range(&self) -> u32 {
+        self.range
+    }
+
+    pub fn code(&self) -> u32 {
+        self.code
+    }
+
+    pub fn buf(&mut self) -> io::Result<&[u8]> {
+        self.stream.fill_buf()
+    }
```

Comment:
I think making `stream` public would also be more readable. A function named `buf()` doesn't really make things more readable than the original `fill_buf`.